### DISCUSSION TITLE
Implement spherical prjection for cameras

### DISCUSSION THREAD

**Discussion by CZakarya** [godotengine/godot-proposals#6849] [3D]
### Describe the project you are working on

I know you might be able to do this with shaders on the camera, but I would like it to be a feature in the editor.

This could be useful for games that have custom VR implementation (like for a user-made system that doesn't use openvr), or odd games that want to have a very weird looking perspective that is still playable (I won't call orthogonal and frustum "playable"). Or if you want to make a magic ball that looks into a different area in the game.

### Describe the problem or limitation you are having in your project

I'm not currently working on a game that needs it, but I'm convinced that someone will.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

As I said in the first text box, it can have multiple uses.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In the property editor for a 3d camera, there is the projection value. My proposal is to add an option to the dropdown.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It'd be a complicated shader as you would need to write the projection from scratch. It just seems like something that should be handled by the engine.

### Is there a reason why this should be core and not an add-on in the asset library?

I haven't really experimented with the add-on API, but I don't think it's possible to make one.

**Comment by Calinou** [comment]
- This is already being tracked in https://github.com/godotengine/godot-proposals/issues/3779.
